### PR TITLE
feat: no_std CI integration

### LINKED ISSUE #8716: Check no std in CI
### Describe the feature

Some crates will have no std support, see #8705 

Currently we don't check that these actually compile with `--no-default-features` (i.e. no std) in CI, but we should.

For resolving this issue, add a `cargo check` pass for all crates in the workspace with `--no-default-features`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by JackG-eth**
Motivation

Closes #8716 

Description

Currently we don't check that these actually compile with --no-default-features (i.e. no std) in CI, but we should.

### CODE REVIEW

**Comment by JackG-eth** [comment]
Hi, Few notes on this:

1. Should this be in the integration.yml or entirely seperated.
2. Should this include other crates like alloy/revm or be member specific
3. Should this include `riscv32imac-unknown-none-elf` in the build script?

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -78,6 +78,21 @@ jobs:
             -- db get static-file headers 100000 \
             | grep 0x91c90676cab257a59cd956d7cb0bceb9b1a71d79755c23c7277a0697ccfaf8c4
 
+  no-std:
```
Comment (DaniPopes) [inline .github/workflows/integration.yml:81]: pls move to lint.yml

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -149,10 +149,35 @@ jobs:
...
+        grafana,
+      ]
     timeout-minutes: 30
     steps:
       - name: Decide whether the needed jobs succeeded or failed
         uses: re-actors/alls-green@release/v1
         with:
           jobs: ${{ toJSON(needs) }}
+
+  no-std:
```
Comment (DaniPopes) [inline .github/workflows/lint.yml:170]: pls move above to after clippy

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -149,10 +149,35 @@ jobs:
     name: lint success
     runs-on: ubuntu-latest
     if: always()
-    needs: [clippy-binaries, clippy, crate-checks, docs, fmt, book, codespell, grafana]
+    needs:
+      [
+        clippy-binaries,
```
Comment (DaniPopes) [inline .github/workflows/lint.yml:154]: can convert to a yaml list (use dashes and remove commas)
```
- a
- b
- c
```

**Review by DaniPopes — CHANGES_REQUESTED** [review]

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**